File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/search/aggregations/bucket/Histogram.java
/*
 * Copyright 2018 the original author or the original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@DataObject
public class Histogram extends MultiBucketsAggregation {

    public Histogram(MultiBucketsAggregation other) {
        super(other);
    }

    /**
     * Build a new <code>Histogram</code> from a <code>JsonObject</code>.
     *
     * @param json the <code>JsonObject</code>
     */
    public Histogram(JsonObject json) {
        super(json);
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/search/aggregations/bucket/Histogram.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.search.aggregations.bucket

import fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket.Histogram
import fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket.Bucket

fun Histogram(
        buckets: Map<String, fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket.Bucket>? = null,
        data: io.vertx.core.json.JsonObject? = null,
        metaData: io.vertx.core.json.JsonObject? = null,
        name: String? = null,
        type: String? = null): Histogram = fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket.Histogram().apply {

    if (buckets != null) {
        this.setBuckets(buckets)
    }
    if (data != null) {
        this.setData(data)
    }
    if (metaData != null) {
        this.setMetaData(metaData)
    }
    if (name != null) {
        this.setName(name)
    }
    if (type != null) {
        this.setType(type)
    }
}

